Userland/Libraries/LibWeb/IndexedDB/Internal/ConnectionQueueHandler.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024, stelar7 <dudedbz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <LibJS/Heap/GCPtr.h>
+#include <LibWeb/DOM/EventTarget.h>
+#include <LibWeb/IndexedDB/IDBRequest.h>
+#include <LibWeb/StorageAPI/StorageKey.h>
+
+namespace Web::IndexedDB {
+
+using ConnectionQueue = AK::Vector<JS::NonnullGCPtr<IDBRequest>>;
```

Comment:
Who visits the queues? If the answer is "nobody", we need to store handles instead of gcptrs.